### PR TITLE
Add resource deletion protection user guide

### PR DESCRIPTION
**Pull request by Vacant2333**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
add the user guide for resource deletion protection

**Which issue(s) this PR fixes**:
Part of https://github.com/karmada-io/karmada/issues/3728

### CODE REVIEW

**Comment by Vacant2333** [comment]
request review~~
/cc @XiShanYongYe-Chang @RainbowMango

**Comment by XiShanYongYe-Chang** [comment]
Thanks @Vacant2333 
/assign

File: docs/administrator/configuration/resource-deletion-protection.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+```
+[root@cluster1]# kubectl delete namespaces minio
+Error from server (Forbidden): admission webhook "resourcedeletionprotection.karmada.io" denied the request: This resource is protected, please make sure to remove the label: resourcetemplate.karmada.io/deletion-protected
+```
+
+## Unprotecting a Resource
+
+If you want to remove Karmada's protection from a resource, you only need to remove the `resourcetemplate.karmada.io/deletion-protected` Label.
+
+Alternatively, you can directly change its value to a value other than `Always`, such as `Never`.
```
Comment (XiShanYongYe-Chang) [inline docs/administrator/configuration/resource-deletion-protection.md:32]: In addition to modifying the label value, can users directly delete the label to achieve the purpose? If yes, we can explain this.

File: docs/administrator/configuration/resource-deletion-protection.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+When you attempt to delete the protected `minio` Namespace, you will see the following output:
+```
+[root@cluster1]# kubectl delete namespaces minio
+Error from server (Forbidden): admission webhook "resourcedeletionprotection.karmada.io" denied the request: This resource is protected, please make sure to remove the label: resourcetemplate.karmada.io/deletion-protected
+```
+
+## Unprotecting a Resource
+
+If you want to remove Karmada's protection from a resource, you only need to **remove** the `resourcetemplate.karmada.io/deletion-protected` Label.
```
Comment (RainbowMango) [inline docs/administrator/configuration/resource-deletion-protection.md:30]: Add an example of removing the label? such as:
```bash
kubectl label namespaces minio resourcetemplate.karmada.io/deletion-protected-
```

**Review by RainbowMango — COMMENTED** [review]
Generally looks good to me. Thanks!

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

Awesome! Thanks @Vacant2333

**Pull request merged**